public/pages/utils/anomalyResultUtils.ts

```diff
@@ -84,21 +89,57 @@ export const getAnomalyResultsWithDateRange = (
       fieldName: AD_DOC_FIELDS.DATA_START_TIME,
     },
   };
-  dispatch(getDetectorResults(detectorId, updatedParams));
+};
+
+export const getAnomalyResultsWithDateRange = (
+  dispatch: Dispatch<any>,
+  startTime: number,
+  endTime: number,
+  detectorId: string
+) => {
+  const params = buildParamsForGetAnomalyResultsWithDateRange(
+    startTime,
+    endTime
+  );
+  dispatch(getDetectorResults(detectorId, params));
 };
 
 const MAX_DATA_POINTS = 1000;
+const MAX_FEATURE_ANNOTATIONS = 100;
```

Comment:
why not have `MAX_FEATURE_ANNOTATIONS` be the same amount as `MAX_DATA_POINTS`?

Reply:
1000 annotations are too crowded on chart. Using 100 looks good

Reply:
If there's any missing feature data points, we want to show this missing feature annotation alert, right? So if there's >100 missing points, wouldn't we miss some annotations that should be there?

Reply:
Yes. Extra missing points won't be shown in that case. Instead, I add sampling on all the missing points, and annotations are generated from those samples. Those annotations can cover all the missing points

Reply:
Got it. Makes more sense looking through the code again. Thanks!